### PR TITLE
Automatic search after 3 characters

### PR DESCRIPTION
**Pull request by jennymakki**
In the filter function the user can now type 3 character before it automatically searches for a room.

### CODE REVIEW

**Comment by jennymakki** [comment]
It looks like I changed everything in the code a few times but that is just because I failed some attempts and wasn't able to reset the code. So I copied the js code from the main branch into this current branch. Sorry for how it looks. I know it makes it harder to review.

**Review by Tobias-hubs — APPROVED** [review]
Good work, a lite hard to review as you said but clever way to go about that problem by copying the code. Nothing wrong with that. Tested it in codespace and it works good :)

File: filter.js [hunk 1]
```diff
@@ -4,106 +4,114 @@ import { ApiArray as array } from "./main.js";
...
+    ).map((checkbox) => checkbox.value);
+
+    // Get all checked checkboxes for tags
+    const checkedTagCheckboxes = Array.from(
+        filterWindow.querySelectorAll('input[name="tags"]:checked')
+    );
+    const selectedTags = checkedTagCheckboxes.map((checkbox) => checkbox.value);
+
+    // Get the selected min and max rating values (from radio buttons)
+    const minRating = filterWindow.querySelector('input[name="minRating"]:checked')?.value;
```
Comment (marcusmoller97) [inline filter.js:30]: Chaining was new for me. But I think it gives a good structure to the code.

**Review by marcusmoller97 — APPROVED** [review]
Loooks good I have nothing to point out. Learned a lot from the code. Keep up the good work.

**Review by Jorlindstrom — APPROVED** [review]
Works very well

File: filter.js [hunk 2]
```diff
@@ -4,106 +4,114 @@ import { ApiArray as array } from "./main.js";
...
+    const maxRating = filterWindow.querySelector('input[name="maxRating"]:checked')?.value;
+
+    // Apply filter logic
+    const filterData = array.filter((card) => {
+        let matches = true;
+
```
Comment (Klumpish) [inline filter.js:36]: great fix, it works great

**Review by Klumpish — APPROVED** [review]

**Pull request merged**